notesapi/v1/search_indexes/documents/note.py

```diff
@@ -0,0 +1,41 @@
+import json
+
+from django.conf import settings
+from django_elasticsearch_dsl import Document, fields, Index
+
+from notesapi.v1.models import Note
+from .analyzers import case_insensitive_keyword, html_strip
+
+__all__ = ('NoteDocument',)
+
+NOTE_INDEX = Index(settings.ELASTICSEARCH_INDEX_NAMES[__name__])
```

Comment:
Unfortunately, we cannot just write `note`. It's too easy 😂
To be honest, it's all about the complex configurations that edx has. There are devstack.py, productions.py, test.py, aws.py, etc. For each level, the index name should also be different. This was done before me and I continue this philosophy. Different prefix (ending) should be. There are different ways to solve this problem, this one seemed to me the most elegant and beautiful.
This is how it will be in the configurations.
```
ELASTICSEARCH_INDEX_NAMES = {'notesapi.v1.search_indexes.documents.note': 'notes_index_test'}
```
```
ELASTICSEARCH_INDEX_NAMES = {'notesapi.v1.search_indexes.documents.note': 'edx_notes_api'}
```
those. the code does not know what the index is called, only configurations knows.

Reply:
Got it! Thanks for the explanation.